tests/src/strategy.spec.ts

```diff
@@ -3,16 +3,61 @@
  * with very little code change. Most of the test cases
  * are already covered.
  */
-import { testData, setupExpressApp } from '../helpers/setupApp';
+import { setupExpressApp } from '../helpers/setupApp';
 import supertest from 'supertest';
 import passport from 'passport';
+import SnykOAuth2Strategy from '../../src';
+
+/**
+ * Data required fro Snyk OAuth2 Strategy.
+ * It is used in other places, so stored in
+ * a separate constant.
+ */
+export const testData = {
+  authorizationURL: 'https://example_authorization.com/authorize',
+  tokenURL: 'https://example_token.com/token',
+  clientID: 'test',
+  clientSecret: 'test',
+  callbackURL: 'https://example_callback.com',
+  scope: 'apps:beta',
+  nonce: 'some_nonce_value',
+};
 
 describe('Strategy with passport', () => {
-  const app = setupExpressApp();
+  const snykOAuth2 = new SnykOAuth2Strategy(
+    {
+      authorizationURL: testData.authorizationURL,
+      tokenURL: testData.tokenURL,
+      clientID: testData.clientID,
+      clientSecret: testData.clientSecret,
+      callbackURL: testData.callbackURL,
+      scope: testData.scope,
+      scopeSeparator: ' ',
+      state: true,
+      passReqToCallback: true,
+      nonce: testData.nonce,
+      // profileFunc,
+    },
+    // Callback function called with the
+    // data fetched as part of authentication
+    (done: any) => {
+      // Notify passport that all work, like storing
+      // of data in DB has been completed
+      done(null, 'Done!');
+    },
+  );
+  const app = setupExpressApp(snykOAuth2);
   const request = supertest(app);
 
   // This will trigger the auth flow
-  app.get('/auth', passport.authenticate('snyk-oauth2', { state: 'test' }));
+  app.get(
+    '/auth',
+    (req, res, next) => {
+      snykOAuth2.nonce = testData.nonce;
+      next();
+    },
+    passport.authenticate('snyk-oauth2', { state: 'test' }),
```

Comment:
Hmmmm, what if `passport.authenticate` has any async things internally, and between the nonce being retrieved from the `snykOAuth2` instance and `passport.authenticate` being called you have another concurrent request coming in? This screams race condition to me 🤔

Reply:
The other solution would be to still pass it into `passport.authenticate`, but either:
- Amend Passport.js to be able to pass this information.
- Export our own wrapper around `passport.authenticate` when we distribute our strategy that accepts the `nonce` value.

Reply:
I can see the problem, but I don't see any other of doing it without updating the `passport` code. There is another hackish solution where you can just pass the `nonce` and `tenant_id` values using `state`. Something like `state: `state-value&nonce=${nonce}&tenant_id=${tenantId}`'

Reply:
Waaaait, I'm working on an assumption that Passport calls the `authorizationParams` function, but I don't see it doing it o.O

Reply:
I think I found a way of doing it. We need to pass on the options we get in the `authorizationParams` as you see the passport code returns an empty object. After doing that the solution you suggested should work.

Reply:
There's also the `tokenParams` call: https://github.dev/jaredhanson/passport-oauth2/blob/ee3fe9f17c0f3a90f2d9d938f267e9942b9fba49/lib/strategy.js#L168

Reply:
We can forwards option on both 🙂

Reply:
Here is the PR https://github.com/snyk/passport-snyk-oauth2/pull/8 that should resolve it. Thanks for the inputs 🙇